db/migrate/20170109221226_create_configuration_projects.rb

```diff
@@ -0,0 +1,17 @@
+class CreateConfigurationProjects < ActiveRecord::Migration[5.0]
+  def change
+    create_table :configuration_projects do |t|
+      t.belongs_to  :configuration_manager, :type => :bigint
+      t.string      :manager_ref
+      t.string      :name
+      t.string      :description
+      t.string      :scm_type
+      t.bigint      :scm_reference_id
```

Comment:
`scm_reference` seems sufficient to me.

It covers git and mercurial branches as well as SVN revision numbers.